internal/dag/httpproxy_processor.go

```diff
@@ -400,22 +401,34 @@ func (p *HTTPProxyProcessor) computeRoutes(
 			namespace = proxy.Namespace
 		}
 
+		if err := pathMatchConditionsValid(include.Conditions); err != nil {
+			validCond.AddErrorf(contour_api_v1.ConditionTypeIncludeError, "PathMatchConditionsNotValid",
+				"include: %s", err)
+			continue
+		}
+
 		includedProxy, ok := p.source.httpproxies[types.NamespacedName{Name: include.Name, Namespace: namespace}]
 		if !ok {
 			validCond.AddErrorf(contour_api_v1.ConditionTypeIncludeError, "IncludeNotFound",
 				"include %s/%s not found", namespace, include.Name)
-			return nil
+
+			// Set 502 response when include was not found but route prefix was valid.
+			if len(include.Conditions) > 0 {
```

Comment:
We should probably require some set of conditions on this, but that's a different issue. =)

Reply:
Here my thinking was that I move the `pathMatchConditionsValid()` call up, so that when we recognize missing include, I know that prefix is valid -> program `502` for the prefix.

But I now realized this is bit more complex:  `pathMatchConditionsValid()`  does not guarantee there is prefix. Should we program `502` also for header conditions if someone used them in `includes[].conditions[]`? I suppose we should?

The code that processes includes has always validated only prefix by calling `pathMatchConditionsValid(include.Conditions)`, there is no call to `headerMatchConditionsValid(include.Conditions)`.
If I can leave prefix field empty and add `includes[].conditions[].headers[]` instead, which gets merged in child proxy route `conditions`.  Lets say for example:

* root has header condition `x-foo` equals `bar`
* child has prefix condition `/api`

both are then combined when processing child proxy's route conditions. In this case traffic is forwarded only if prefix is `/api` and request has `x-foo` header with value `bar`. I guess this is intentional?

Reply:
>  In this case traffic is forwarded only if prefix is /api and request has x-foo header with value bar. I guess this is intentional?

Yup that's how it's designed, the resulting set of conditions define the match so adding in different headers could let the request route to different places.

Reply:
Now both prefix and header conditions are considered when programming 502 response for include.

There was an existing test case `TestDAGStatus/duplicate_include_condition_headers` that had header (intentionally) invalid header conditions for includes. Because header conditions for includes were not previously validated at root proxy, they were propagated to child and child got blamed for the error.  Now the error condition is set for root proxy where the invalid conditions are and the test case is fixed as well.